harness/determined/_trial_context.py

```diff
@@ -14,10 +14,12 @@ class TrialContext:
     def __init__(
         self,
         core_context: core.Context,
-        env: det.EnvContext,
+        env: Optional[det.EnvContext] = None,
```

Comment:
how do you feel about removing the `env` object entirely?

It's basically a pure tech debt item.  In local training mode, it's full of phony values that we make up to do local training.  I would like to see the `env` object not be found anywhere in our pytorch code when the pytorch trainer API lands.

I am not saying you need to do it now before we've finish iterating on the API, of course.  But I think it's important to do in the "implementation and automated testing" milestone, because I suspect there will be things in env that end up needing to be passed in by the user in `pytorch.init()` or something... which is important we don't release an API and figure that out later.

Reply:
oh, and definitely I'm not proposing any modifications to EstimatorTrialContext or TFKerasTrialContext.  IMO those can continue taking an `env` as part of their subclass definition.

Reply:
or... it looks like you already removed all env references from pytorchtrial and pytorchcontext?  If that's the case, then you're like 90% of the way there.

Last step is make sure there's not even a `self.env` that other developers might be tempted to use, ha